sdk/storage/Azure.Storage.Blobs/src/PageBlobClient.cs

```diff
@@ -2887,6 +2887,224 @@ public new PageBlobClient WithVersion(string versionId)
+                size: size,
+                options: options,
+                async: true,
+                cancellationToken: cancellationToken)
+                .ConfigureAwait(false);
+
+        /// <summary>
+        /// Opens a stream for writing to the blob.
+        /// </summary>
+        /// <param name="overwrite">
+        /// Whether an existing blob should be deleted and recreated.
+        /// </param>
+        /// <param name="position">
+        /// The offset within the page blob to begin writing from.
+        /// </param>
+        /// <param name="size">
+        /// Required if overwrite is set to true, or the underlying
+        /// Page Blob is being created for the first time.
+        /// Specifies the size of the new Page Blob.
+        /// </param>
+        /// <param name="options">
+        /// Optional parameters.
+        /// </param>
+        /// <param name="async">
+        /// Whether to invoke the operation asynchronously.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A stream to write to the Append Blob.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        private async Task<Stream> OpenWriteInternal(
+            bool overwrite,
+            long position,
+            long? size,
+            PageBlobOpenWriteOptions options,
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            DiagnosticScope scope = ClientDiagnostics.CreateScope($"{nameof(PageBlobClient)}.{nameof(OpenWrite)}");
+
+            try
+            {
+                scope.Start();
+
+                ETag? etag;
+
+                if (overwrite)
+                {
+                    if (size == null)
+                    {
+                        throw new ArgumentException($"{nameof(size)} must be set if {nameof(overwrite)} is set to true");
+                    }
+
+                    Response<BlobContentInfo> createResponse = await CreateInternal(
+                        size: size.Value,
+                        sequenceNumber: default,
+                        httpHeaders: default,
+                        metadata: default,
+                        tags: default,
+                        conditions: options?.OpenConditions,
+                        async: async,
+                        cancellationToken: cancellationToken)
+                        .ConfigureAwait(false);
+
+                    etag = createResponse.Value.ETag;
+                }
+                else
+                {
+                    try
+                    {
+                        Response<BlobProperties> propertiesResponse = await GetPropertiesInternal(
```

Comment:
`CreateIfNotExists()` wipes out the user's `AccessConditions`, so I can't use it here.